src/clearskies/column_types/created_micro.py

```diff
@@ -8,8 +8,8 @@ class CreatedMicro(DateTimeMicro):
         "utc",
     ]
 
-    def __init__(self, di, datetime):
-        super().__init__(di)
+    def __init__(self, di, datetime, timezone):
```

Comment:
Why pass the timezone in the constructor?

Reply:
I created in the the standard dependencies `provide_timezone` 

```python
def provide_timezone(self):
        """Set the default timezone."""
        import datetime
        return datetime.UTC
```
In this way you can still set the different timezone global, without the need to set it on on each column

Reply:
(Sorry, after I made my initial comment I realized I had more thoughts)

I would normally think of this kind of thing as a matter of configuration, so it would be set through the column config instead of the constructor.  For it to come through the column config means that you  would set it via the usual:

```
def columns_configuration():
    return {
         created_micro('my_column', timezone=datetime.timezone.utc),
    }
```

By setting it as a constructor argument it is no longer configurable at the column level but instead must be set on the dependency injection container, so it will be constant for all datetime columns.  That also means that we should set a default in the dependency injection system otherwise people will have to add it to their apps or all the datetime columns will break.  That's easy though.  I'd just add it to the [standard dependencies](https://github.com/cmancone/clearskies/blob/master/src/clearskies/di/standard_dependencies.py) with something like:

```
def provide_timezone(self):
    import datetime
    return datetime.timezone.utc
```

Which, now that I say it out loud, makes a lot of sense.  It seems unlikely that people are going to change timezones from one column to another.  Most likely things will always be operating in one timezone (and that timezone will be UTC), so I like the idea of having a configuration to  set the timezone of the whole application.

Then again, their might be some cases where someone does need date columns in separate timezones (maybe they are working with an external system that is *not* set to UTC).  If that need ever comes up though, I could also add in a timezone at the column config level to override the application-wide default.

So: yeah, let's go for it.  However, we need to provide the timezone variable through the DI system or we'll break everyone's applications.

Reply:
There already some overrides in place, although they have same outcome at the moment:

```python
def pre_save(self, data, model):
        if model.exists:
            return data
        if self.config("utc", silent=True):
            now = self.datetime.datetime.now(self.datetime.timezone.utc)
        else:
            now = self.datetime.datetime.now(self._timezone)
        return {**data, self.name: now}
```
which could be easily extended with an extra config value like `timezone` for example
```python
def pre_save(self, data, model):
        if model.exists:
            return data
        if self.config("utc", silent=True):
            now = self.datetime.datetime.now(self.datetime.timezone.utc)
        elif self.config("timezone", silent=True):
            now = self.datetime.datetime.now(self.config("timezone"))
        else:
            now = self.datetime.datetime.now(self._timezone)
        return {**data, self.name: now}
```